### PR TITLE
enh(driver): tmq async commit callback

### PR DESCRIPTION
**Pull request by huolibo**

### CODE REVIEW

**Review by zyyang-taosdata — APPROVED** [review]
LGTM

File: source/client/src/clientTmqConnector.c [hunk 1]
```diff
@@ -17,6 +17,36 @@
...
+    case 2:
+      return;
+  }
+
+  if (g_vm == NULL) {
+    (*env)->GetJavaVM(env, &g_vm);
+  }
+
+  jclass offset = (*env)->FindClass(env, "com/taosdata/jdbc/tmq/OffsetWaitCallback");
+  jclass g_offsetCallbackClass = (*env)->NewGlobalRef(env, offset);
```
Comment (sangshuduo) [inline source/client/src/clientTmqConnector.c:41]: should we check if offset is accessible?

File: source/client/src/clientTmqConnector.c [hunk 2]
```diff
@@ -17,6 +17,36 @@
...
+  g_offsetCallback = (*env)->GetMethodID(env, g_offsetCallbackClass, "commitCallbackHandler", "(I)V");
+  (*env)->DeleteLocalRef(env, offset);
```
Comment (sangshuduo) [inline source/client/src/clientTmqConnector.c:43]: same worry

**Review by sangshuduo — APPROVED** [review]
LGTM

**Pull request merged**